scipy/signal/_splines.py

```diff
@@ -0,0 +1,156 @@
+        The input signal. If 2D, then the filter will be applied in a batched
+        fashion across the last axis.
+    c0, z1 : scalar
+        Parameters in the transfer function.
+    precision :
+        Specifies the precision for calculating initial conditions
+        of the recursive filter based on mirror-symmetric input.
+
+    Returns
+    -------
+    output : ndarray
+        The filtered signal.
+    """
+    if np.abs(z1) >= 1:
+        raise ValueError('|z1| must be less than 1.0')
+
+    if signal.ndim > 2:
+        raise ValueError('Input must be 1D or 2D')
+
+    squeeze_dim = False
+    if signal.ndim == 1:
+        signal = signal[None, :]
+        squeeze_dim = True
+
+    y0 = symiirorder1_ic(signal, z1, precision)
+
+    # Apply first the system 1 / (1 - z1 * z^-1)
+    b = np.ones(1, dtype=signal.dtype)
+    a = np.r_[1, -z1]
+    a = a.astype(signal.dtype)
+
+    # Compute the initial state for lfilter.
+    zii = y0 * z1
+
+    y1, _ = lfilter(b, a, axis_slice(signal, 1), zi=zii)
+    y1 = np.c_[y0, y1]
+
+    # Compute backward symmetric condition and apply the system
+    # c0 / (1 - z1 * z)
+    b = np.asarray([c0], dtype=signal.dtype)
+    out_last = -c0 / (z1 - 1.0) * axis_slice(y1, -1)
+
+    # Compute the initial state for lfilter.
+    zii = out_last * z1
+
+    out, _ = lfilter(b, a, axis_slice(y1, -2, step=-1), zi=zii)
+    out = np.c_[axis_reverse(out), out_last]
+
+    if squeeze_dim:
+        out = out[0]
+
+    return out
+
+
+def symiirorder2(input, r, omega, precision=-1.0):
+    """
+    Implement a smoothing IIR filter with mirror-symmetric boundary conditions
+    using a cascade of second-order sections.  The second section uses a
+    reversed sequence.  This implements the following transfer function::
+
+                                  cs^2
+         H(z) = ---------------------------------------
+                (1 - a2/z - a3/z^2) (1 - a2 z - a3 z^2 )
+
+    where::
+          a2 = 2 * r * cos(omega)
+          a3 = - r ** 2
+          cs = 1 - 2 * r * cos(omega) + r ** 2
+
+    Parameters
+    ----------
+    input : ndarray
+        The input signal.
+    r, omega : float
+        Parameters in the transfer function.
+    precision : float
+        Specifies the precision for calculating initial conditions
+        of the recursive filter based on mirror-symmetric input.
+
+    Returns
+    -------
+    output : ndarray
+        The filtered signal.
+    """
+    if r >= 1.0:
+        raise ValueError('r must be less than 1.0')
+
+    if input.ndim > 2:
+        raise ValueError('Input must be 1D or 2D')
+
+    if not input.flags.c_contiguous:
+        input = input.copy()
+
+    squeeze_dim = False
+    if input.ndim == 1:
+        input = input[None, :]
+        squeeze_dim = True
+
+    rsq = r * r
+    a2 = 2 * r * np.cos(omega)
+    a3 = -rsq
+    cs = np.atleast_1d(1 - 2 * r * np.cos(omega) + rsq)
+    sos = np.atleast_2d(np.r_[cs, 0, 0, 1, -a2, -a3]).astype(input.dtype)
```

Comment:
Is this correct for `input`s of integer dtype?

Reply:
Let me check